### PR TITLE
chore: replace tslint with eslint

### PR DESCRIPTION
**Pull request by dbjorge**
#### Description of changes

This PR replaces all usage of tslint with eslint, per current SDL requirements and tslint's deprecation. I started from a clean eslint configuration using the SDL recommendations, rather than keeping the existing tslint rule suppressions, since this codebase is small enough to just fix everything during the conversion.

#### Pull request checklist

- [n/a] Addresses an existing issue: Fixes #0000
- [x] Added relevant unit test for your changes. (`yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: src/progress-reporter/check-run/check-run-creator.spec.ts [hunk 1]
```diff
@@ -114,20 +112,19 @@ describe(CheckRunCreator, () => {
                 title: checkRunDetailsTitle,
                 summary: disclaimerText,
                 annotations: [],
-                text: stripIndent`
-                ${message}`,
+                text: stubErrorMarkdown,
```
Comment (peterdur) [inline src/progress-reporter/check-run/check-run-creator.spec.ts:115]: No longer need `stripIndent` import above after this change

Comment (peterdur) [reply src/progress-reporter/check-run/check-run-creator.spec.ts:115]: (and then can remove packages `common-tags` and `@types/common-tags`

Comment (dbjorge) [reply src/progress-reporter/check-run/check-run-creator.spec.ts:115]: Leaving no-unused-vars enabled in the tests would have actually caught this, hah! I think I'll re-enable it and do inline suppressions

**Review by peterdur — APPROVED** [review]
Looks good!

**Pull request merged**